src/cmd/services/m3coordinator/downsample/samples_appender.go

```diff
@@ -52,6 +56,40 @@ func (a samplesAppender) AppendGaugeSample(value float64) error {
 	return a.agg.AddUntimed(sample, a.stagedMetadatas)
 }
 
+func (a *samplesAppender) AppendCounterTimedSample(value int64, t time.Time) error {
```

Comment:
Super nit but Datapoint structs are usually:

```
type Datapoint struct {
    Timestamp time.Time
     value int64
}
```

Which makes me think time should proceed value in the API

Reply:
Sure thing, sounds good.